Release/src/websockets/client/ws_client_wspp.cpp

```diff
@@ -207,11 +209,30 @@ class wspp_callback_client : public websocket_client_callback_impl, public std::
                     }
                     if(m_openssl_failed)
                     {
-                        return http::client::details::verify_cert_chain_platform_specific(verifyCtx, utility::conversions::to_utf8string(m_uri.host()));
+
+                        if (!http::client::details::is_end_certificate_in_chain(verifyCtx))
+                        {
+                            // Continue until we get the end certificate.
+                            return true;
+                        }
+
+                        auto chainFunc = [this](const std::shared_ptr<http::client::certificate_info>& cert_info) {
+                            return m_config.invoke_certificate_chain_callback(cert_info);
+                        };
+
+                        return http::client::details::verify_cert_chain_platform_specific(verifyCtx, utility::conversions::to_utf8string(m_uri.host()), chainFunc);
                     }
 #endif
                     boost::asio::ssl::rfc2818_verification rfc2818(utility::conversions::to_utf8string(m_uri.host()));
-                    return rfc2818(preverified, verifyCtx);
+                    if (!rfc2818(preverified, verifyCtx))
+                    {
+                        return false;
+                    }
+
+                    auto info = std::make_shared<http::client::certificate_info>(utility::conversions::to_utf8string(m_uri.host()), get_X509_cert_chain_encoded_data(verifyCtx));
+                    info->verified = true;
+
+                    return m_config.invoke_certificate_chain_callback(info);
```

Comment:
Wild guess: I would expect that this would also be invoked more than once per connection on Linux as it seems to be duplicated code referred to my comment above.

Reply:
Correct as the check "is_end_certificate_in_chain" isn't done on Linux.